nav2_msgs/srv/LoadMap.srv

```diff
@@ -0,0 +1,23 @@
+# Type definitions
+uint8 TYPE_FILE=0
+uint8 TYPE_URL=1
+
+# Type of map resource
+uint8 type
```

Comment:
Let's try to coordinate with my ROS 1 PR (https://github.com/ros/common_msgs/pull/152) to make bridging in the future more straight forward.

A path to the file can also be represented as a URL. I think doing something similar to what I've done in https://github.com/ros/common_msgs/pull/152 leaves to door open to other locator types without having to modify the message definition.

Reply:
jacob, how do you plan to get the YAML file from the URL? Are you planning to use `curl` or is there an easier way to load a YAML file directly from a URL?

Reply:
I haven't thought about it too much. For the file URL, `curl` might work. As a first iteration, it might also be just as easy to write (or find) a simple URL parser to check if the prefix of the URL is `file://` and then take the rest of the URL as the path to the YAML file.

To load a YAML from a package directory (eg. `package://`) we'll need URL parser anyways to get the package name and relative path, and then we can use the ament API to locate the file.

Reply:
Actually, we should probably use the existing [resource_retriever](https://github.com/ros/resource_retriever) for the job. It it a curl-like tool that also supports `package://` :)

I forgot about it.

Reply:
OK, that's very helpful. I wasn't aware of resource_retriever.

I'll have to re-write part of this PR to use that and the message definition you created.